Description: Fix up cmake Ada support for compatibility with cmake 2.8.9 
Author: Andrew Ross <andrewross@users.sourceforge.net>
Last-Update: 2012-07-19

Index: plplot-5.9.9/cmake/modules/language_support/cmake/CMakeAdaInformation.cmake
===================================================================
--- plplot-5.9.9.orig/cmake/modules/language_support/cmake/CMakeAdaInformation.cmake	2011-10-13 01:43:01.000000000 +0100
+++ plplot-5.9.9/cmake/modules/language_support/cmake/CMakeAdaInformation.cmake	2012-07-19 23:03:27.835197249 +0100
@@ -29,6 +29,18 @@
   SET(CMAKE_SHARED_LIBRARY_CREATE_Ada_FLAGS ${CMAKE_SHARED_LIBRARY_CREATE_C_FLAGS})
 ENDIF(NOT CMAKE_SHARED_LIBRARY_CREATE_Ada_FLAGS)
 
+IF(NOT CMAKE_Ada_COMPILE_OPTIONS_PIC)
+  SET(CMAKE_Ada_COMPILE_OPTIONS_PIC ${CMAKE_C_COMPILE_OPTIONS_PIC})
+ENDIF(NOT CMAKE_Ada_COMPILE_OPTIONS_PIC)
+
+IF(NOT CMAKE_Ada_COMPILE_OPTIONS_PIE)
+  SET(CMAKE_Ada_COMPILE_OPTIONS_PIE ${CMAKE_C_COMPILE_OPTIONS_PIE})
+ENDIF(NOT CMAKE_Ada_COMPILE_OPTIONS_PIE)
+
+IF(NOT CMAKE_Ada_COMPILE_OPTIONS_DLL)
+  SET(CMAKE_Ada_COMPILE_OPTIONS_DLL ${CMAKE_C_COMPILE_OPTIONS_DLL})
+ENDIF(NOT CMAKE_Ada_COMPILE_OPTIONS_DLL)
+
 IF(NOT CMAKE_SHARED_LIBRARY_Ada_FLAGS)
   SET(CMAKE_SHARED_LIBRARY_Ada_FLAGS ${CMAKE_SHARED_LIBRARY_C_FLAGS})
 ENDIF(NOT CMAKE_SHARED_LIBRARY_Ada_FLAGS)
Index: plplot-5.9.9/cmake/modules/language_support/cmake/CMakeDetermineAdaCompiler.cmake
===================================================================
--- plplot-5.9.9.orig/cmake/modules/language_support/cmake/CMakeDetermineAdaCompiler.cmake	2011-10-13 01:43:01.000000000 +0100
+++ plplot-5.9.9/cmake/modules/language_support/cmake/CMakeDetermineAdaCompiler.cmake	2013-06-25 16:55:17.698208176 +0100
@@ -55,8 +55,11 @@
 
 # configure variables set in this file for fast reload later on
 #CONFIGURE_FILE(${CMAKE_ROOT}/Modules/CMakeAdaCompiler.cmake.in 
+#message(STATUS "DEBUG: CMAKE_BINARY_DIR = ${CMAKE_BINARY_DIR}")
+#message(STATUS "DEBUG: CMAKE_FILES_DIRECTORY = ${CMAKE_FILES_DIRECTORY}")
+#message(STATUS "DEBUG: CMAKE_PLATFORM_INFO_DIR = ${CMAKE_PLATFORM_INFO_DIR}")
 CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/cmake/modules/language_support/cmake/CMakeAdaCompiler.cmake.in 
-               "${CMAKE_BINARY_DIR}${CMAKE_FILES_DIRECTORY}/CMakeAdaCompiler.cmake" IMMEDIATE)
+               "${CMAKE_PLATFORM_INFO_DIR}/CMakeAdaCompiler.cmake" IMMEDIATE)
 MARK_AS_ADVANCED(CMAKE_AR)
 
 SET(CMAKE_Ada_COMPILER_ENV_VAR "ADA")
